topics/go/algorithms/data/bst/bst.go

```diff
@@ -0,0 +1,113 @@
+package bst
```

Comment:
We need some information about what a binary search tree is.  Add a doc.go and provide some information.